core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -624,7 +627,13 @@ private[spark] class TaskSchedulerImpl(
         }
 
         if (!launchedAnyTask) {
-          taskSet.getCompletelyExcludedTaskIfAny(hostToExecutors).foreach { taskIndex =>
+
+          val hostToExecutorsForTaskSet = hostToExecutors.map { case (host, execsOnHost) =>
```

Comment:
We are generating this map each time there are no tasks scheduled ... for larger clusters, this can go into thousands of entries in the map.
Instead, why not maintain this map directly ?

Reply:
that's a good point, we care about time here more.  I think you can get rid of the executorIdToResourceProfileId in that case to so space won't be as bad.